tools/test-proxy/Azure.Sdk.Tools.TestProxy/Common/RecordSession.cs

```diff
@@ -155,6 +165,14 @@ public async Task Sanitize(RecordedTestSanitizer sanitizer, bool shouldLock = tr
                 }
             }
         }
+        private void LogSanitizerModification(string sanitizerId, string entryPreSanitize, string entryPostSanitize)
+        {
+            var result = StringComparer.OrdinalIgnoreCase.Compare(entryPreSanitize, entryPostSanitize);
+            if (result != 0)
+            {
+                DebugLogger.LogInformation($"Sanitizer {sanitizerId} modified the entry\nbefore:\n {entryPreSanitize}\nafter:\n {entryPostSanitize}");
```

Comment:
for custom / user-specified ones can we change the message slightly to indicate the distinction? e.g. `Central sanitizer rule {sanitizerId}` vs `User specified sanitizer rule {sanitizerId}`